server/src/main/java/org/elasticsearch/index/query/SearchExecutionContext.java

```diff
@@ -228,6 +230,15 @@ private void reset() {
         this.nestedScope = new NestedScope();
     }
 
+    // Set alias filter, so it can be applied for queries that need it (e.g. knn query)
+    public void setAliasFilter(QueryBuilder aliasFilter) {
+        this.aliasFilter = aliasFilter;
+    }
+
+    public QueryBuilder getAliasFilter() {
+        return aliasFilter;
+    }
+
```

Comment:
I wonder if `SearchContext` should be made available as part of `rewrite` instead of `SearchExecutionContext` since alias filter is available as part of that? I do concede this may be too large a change for now.

Reply:
Thanks for the feedback, but looks like indeed this would be too large a change